codalab/apps/teams/forms.py

```diff
@@ -89,3 +95,144 @@ def clean(self):
         cleaned_data = super(TeamMembershipForm, self).clean()
 
         return cleaned_data
+
+
+class OrganizerTeamForm(forms.ModelForm):
+
+    text_members = forms.CharField(required=False, widget=forms.Textarea)
+
+    class Meta:
+        model = Team
+        fields = ('description', 'name')
+
+    def __init__(self, *args, **kwargs):
+        self.user_list = []
+        self.competition = Competition.objects.get(pk=kwargs.pop('competition_pk'))
+        self.creator = ClUser.objects.get(pk=kwargs.pop('creator_pk'))
+        super(OrganizerTeamForm, self).__init__(*args, **kwargs)
+        # This check is to see if we have an existing object with a PK, or if we're making a new one
+        # If we have an existing we grab existing members for the form.
+        if hasattr(self, 'instance'):
+            if hasattr(self.instance, 'pk'):
+                if self.instance and self.instance.pk:
+                    team = Team.objects.get(pk=self.instance.pk)
+                    current_members = team.members.all()
+                    name_text_list = ""
+                    name_list = []
+                    for member in current_members:
+                        name_list.append(member.username)
+                        name_text_list = ",".join(name_list)
+                    self.initial['text_members'] = name_text_list
+
+    def clean_text_members(self):
+        if self.cleaned_data.get('text_members'):
+            team_members_text = self.cleaned_data.get('text_members')
+            team_members_list = team_members_text.replace(' ','').split(',')
+            # team_members_list = team_members_text.replace(' ','').split('\n')
+            self.user_list = ClUser.objects.filter(Q(username__in=team_members_list) | Q(email__in=team_members_list))
+
+    def clean_name(self):
+        if self.cleaned_data.get('name'):
+            if hasattr(self, 'instance'):
+                if self.instance.pk:
+                    existing_team_with_name = Team.objects.filter(name=self.cleaned_data.get('name')).exclude(pk=self.instance.pk)
+                    if existing_team_with_name:
+                        raise ValidationError(('Invalid value for name. A team already exists with that name.'),
+                                              code='invalid')
+            else:
+                existing_team_with_name = Team.objects.filter(name=self.cleaned_data.get('name'))
+                if existing_team_with_name:
+                    raise ValidationError(('Invalid value for name. A team already exists with that name.'), code='invalid')
+        return self.cleaned_data.get('name')
+
+    def save(self, commit=True):
+        self.instance.creator = self.creator
+        self.instance.competition = self.competition
+        self.instance.status = TeamStatus.objects.get(codename=TeamStatus.APPROVED)
+
+        # We need to call save before adding members
+        self.instance.save()
```

Comment:
```python
if not self.instance.pk: 
    self.instance.save()```

Maybe instead here? probably not too important

Reply:
This change caused issues. I don't think it hurts too much to call `save` indiscriminately here